pyne/endl.py

```diff
@@ -0,0 +1,239 @@
+
+
+class Library(rx.RxLib):
+    """A class for a file which contains multiple ENDL tables."""
+    def __init__(self, fh):
+        self.mts = {}
+        self.structure = {}
+        self.mat_dict = {}
+        self.intdict = {
+                0: self._linlin,
+                2: self._linlin,
+                3: self._loglin,
+                4: self._linlog,
+                5: self._loglog,
+                }
+        self.fh = fh
+        # read headers for all tables
+        self._read_headers()
+
+#    def load(self):
+#        """load()
+#        Read the ENDL file into a NumPy array.
+#
+#        Returns
+#        -------
+#        data : np.array, 1d, float64
+#            Returns a 1d float64 NumPy array.
+#        """
+#        opened_here = False
+#        if isinstance(self.fh, basestring):
+#            fh = open(self.fh, 'r')
+#            opened_here = True
+#        else:
+#            fh = self.fh
+#        fh.readline()
+#        data = fromendf_tok(fh.read())
+#        fh.seek(0)
+#        if opened_here:
+#            fh.close()
+#        return data
+
+    def _read_headers(self):
+        opened_here = False
+        if isinstance(self.fh, basestring):
+            fh = open(self.fh, 'r')
+            opened_here = True
+        else:
+            fh = self.fh
+
+        while True:
+            # get header lines
+            line1 = fh.readline()
+            line2 = fh.readline()
+
+            # EOF?
+            if not line2:
+                break
+
+            # store the start of the table
+            start = fh.tell()
+
+            # parse the first header line
+            nuc_zzzaaa = line1[0:6].strip()
+            yi = int(line1[7:9].strip() or -1)
+            yo = int(line1[10:12].strip() or -1)
+            aw_str = line1[13:24]
+            aw = utils.endftod(aw_str) if aw_str else np.float64(-1.)
+            date = line1[25:31].strip()
+            iflag = int(line1[31].strip() or 0)
+
+            # parse the second header line
+            rdesc = int(line2[0:2].strip() or -1)
+            rprop = int(line2[2:5].strip() or -1)
+            rmod = int(line2[5:8].strip() or -1)
+            x1_str = line2[21:32]
+            x1 = utils.endftod(x1_str or -1.)
+
+            # convert to Pyne native formats
+            nuc = nucname.zzzaaa_to_id(nuc_zzzaaa)
+
+            # insert the nucleus in the self.structure dictionary
+#            if nuc not in self.structure:
+#                self.structure.update(
+#                        { nuc: {
+#                            'particle_in': [],
+#                            'particle_out': [],
+#                            'data': {},
+#                            'start': start,
+#                            }})
+
+            # skip to the end of the table
+            line = fh.readline()
+            while line and not END_OF_TABLE_RE.match(line):
+                line = fh.readline()
+
+
+
+    def _linlin(self, e_int, xs, low, high):
```

Comment:
> This is one of a few different interpolation functions. Do any of the
already included modules - I'm thinking numpy here - have interpolation
routines already? If not, these should perhaps live in some utility module
somewhere.

NumPy does have interpolations functions but I think that the value here is
that they set the bounds correctly to match what is in the spec.  Some of
the implementation could possibly be cleaned up with that in mind.